config/initializers/active_hash.rb

```diff
@@ -1,10 +1,8 @@
 Rails.application.config.to_prepare do
-  unless defined?(DATA_SOURCES)
-    DATA_SOURCES = {
-      "connectors" => GasAssets::Connector,
-      "pipes"      => GasAssets::Pipe,
-    }.freeze
-  end
+  DATA_SOURCES ||= {
```

Comment:
>  Okay! I added that because Ruby keeps complaining about redefining the constant

Yes, I remember those messages. With this line I also don't see those warnings anymore though. It mostly happened when I had multiple rails consoles open. But right now I can't seem to reproduce it... (when switching back to the old code aswell). Also running `Rails.cache.clear` from the console doesn't seem to have any effect.

Hmm it might be. I'll give it a shot.